File: app/src/main/java/com/pdg/androidbasics/viewcontroller/ListAdapter.java
package com.pdg.androidbasics.viewcontroller;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;
import com.pdg.androidbasics.R;
import com.pdg.androidbasics.model.Item;

import java.util.ArrayList;

public class ListAdapter extends BaseAdapter {

    private final Context context;
    private final ArrayList<Item> values;

    public ListAdapter(Context context, ArrayList<Item> values){

        this.context = context;
        this.values = values;
    }

    @Override
    public Object getItem(int position) {
        return values.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getCount() {
        return values.size();
    }

    @Override
    public boolean hasStableIds() {
        return super.hasStableIds();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);

        View rowView = inflater.inflate(R.layout.list_item_row, parent,false);

        TextView nameTV = rowView.findViewById(R.id.rowNameLabel);
        nameTV.setText(values.get(position).getName());

        TextView subtitleTV = rowView.findViewById(R.id.rowSubtitleLabel);
        subtitleTV.setText(values.get(position).getSubtitle());

        TextView priceTV = rowView.findViewById(R.id.rowPriceLabel);
        priceTV.setText(""+values.get(position).getPrice());

        return rowView;
    }
}

File: app/src/main/java/com/pdg/androidbasics/viewcontroller/ListAdapter.kt
package com.pdg.androidbasics.viewcontroller

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.pdg.androidbasics.R
import com.pdg.androidbasics.model.Item

import java.util.ArrayList

class ListAdapter(private val context: Context, private val values: ArrayList<Item>) : BaseAdapter() {

    override fun getItem(position: Int): Any {
        return values[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getCount(): Int {
        return values.size
    }

    override fun hasStableIds(): Boolean {
        return super.hasStableIds()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
//        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

        val inflater = LayoutInflater.from(context)
        val rowView = inflater.inflate(R.layout.list_item_row, parent, false)

        val nameTV = rowView.findViewById<TextView>(R.id.rowNameLabel)
        nameTV.text = values[position].name

        val subtitleTV = rowView.findViewById<TextView>(R.id.rowSubtitleLabel)
        subtitleTV.text = values[position].subtitle

        val priceTV = rowView.findViewById<TextView>(R.id.rowPriceLabel)
        priceTV.text = "" + values[position].price

        return rowView
    }
}